src/pages/SearchPage/index.js

```diff
@@ -0,0 +1,181 @@
+        setSearchValue(text);
+    };
+
+    const {
+        recentReports,
+        personalDetails: localPersonalDetails,
+        userToInvite,
+        headerMessage,
+    } = useMemo(() => {
+        if (!isScreenTransitionEnd) {
+            return {
+                recentReports: {},
+                personalDetails: {},
+                userToInvite: {},
+                headerMessage: '',
+            };
+        }
+        const options = OptionsListUtils.getSearchOptions(reports, personalDetails, debouncedSearchValue.trim(), betas);
+        const header = OptionsListUtils.getHeaderMessage(options.recentReports.length + options.personalDetails.length !== 0, Boolean(options.userToInvite), debouncedSearchValue);
+        return {...options, headerMessage: header};
+    }, [debouncedSearchValue, reports, personalDetails, betas, isScreenTransitionEnd]);
+
+    const sections = useMemo(() => {
+        const newSections = [];
+        let indexOffset = 0;
+
+        if (recentReports.length > 0) {
+            newSections.push({
+                data: recentReports,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += recentReports.length;
+        }
+
+        if (localPersonalDetails.length > 0) {
+            newSections.push({
+                data: localPersonalDetails,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += recentReports.length;
+        }
+
+        if (userToInvite) {
+            newSections.push({
+                data: [userToInvite],
+                shouldShow: true,
+                indexOffset,
+            });
+        }
+
+        return newSections;
+    }, [localPersonalDetails, recentReports, userToInvite]);
+
+    const selectReport = (option) => {
+        if (!option) {
+            return;
+        }
+
+        if (option.reportID) {
+            setSearchValue('');
+            Navigation.dismissModal(option.reportID);
+        } else {
+            Report.navigateToAndOpenReport([option.login]);
+        }
+    };
+
+    const handleScreenTransitionEnd = () => {
+        setIsScreenTransitionEnd(true);
+    };
+
+    const isOptionsDataReady = useMemo(() => ReportUtils.isReportDataReady() && OptionsListUtils.isPersonalDetailsReady(personalDetails), [personalDetails]);
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            testID={SearchPage.displayName}
+            onEntryTransitionEnd={handleScreenTransitionEnd}
+        >
+            {({didScreenTransitionEnd, safeAreaPaddingBottomStyle}) => (
+                <>
+                    <HeaderWithBackButton title={translate('common.search')} />
+                    <View style={[themeStyles.flex1, themeStyles.w100, safeAreaPaddingBottomStyle]}>
+                        <SelectionList
+                            sections={didScreenTransitionEnd && isOptionsDataReady ? sections : []}
```

Comment:
Can we fallback to a globally defined empty array instead of creating new ones here? I'm talking about the `[]` at the end. Ideally there'd be a one-time-defined `EMPTY_ARRAY` constant of some sort.

Reply:
ha, conveniently there is one defined in `CONST` :)